### PR TITLE
Migrate mozilla importer

### PR DESCRIPTION
**Pull request by TG1999**
Signed-off-by: Tushar Goel <tushar.goel.dav@gmail.com>

### CODE REVIEW

File: vulnerabilities/tests/test_mozilla.py [hunk 1]
```diff
@@ -8,97 +8,20 @@
...
-        )
-
-
-def test_categorize_versions_ranges():
-    # Populate if impacted version is filled
-    pass
+TEST_DATA = os.path.join(BASE_DIR, "test_data/mozilla/")
+
+
+def test_import():
```
Comment (pombredanne) [inline vulnerabilities/tests/test_mozilla.py:19]: These should be two different tests IMHO, not one.
Also you would nee to delete https://github.com/nexB/vulnerablecode/blob/main/vulnerabilities/tests/test_data/mozilla.zip but you may want to make sure we cover all the test cases that were in that zip

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thanks! See some nits for your review.

File: vulnerabilities/tests/test_mozilla.py (hunk 1 above)
Comment (TG1999) [reply vulnerabilities/tests/test_mozilla.py:19]: @pombredanne will separate the tests and delete `mozilla.zip`, in these new tests we are asserting the whole advisory at once and not checking every property one by one.

File: vulnerabilities/importers/mozilla.py [hunk 2]
```diff
@@ -146,18 +136,29 @@ def mfsa_id_from_filename(filename):
...
+            continue
             # pkg is of the form "Firefox ESR 1.21" or "Thunderbird 2.21"
-            name=pkg.rsplit(None, 1)[0],
-            version=pkg.rsplit(None, 1)[1],
-        )
-        for pkg in pkgs
-        if pkg
-    ]
-    return package_urls
+        version = pkg.rsplit(None, 1)[1]
```
Comment (TG1999) [inline vulnerabilities/importers/mozilla.py:144]:
```suggestion
        name, _, version = pkg.rpartition(" ")
```

File: vulnerabilities/importers/mozilla.py [hunk 3]
```diff
@@ -82,47 +83,36 @@ def get_advisories_from_yml(mfsa_id, lines) -> List[AdvisoryData]:
     for cve, advisory in data["advisories"].items():
         # These may contain HTML tags
         summary = BeautifulSoup(advisory.get("description", ""), features="lxml").get_text()
-
-        advisories.append(
-            AdvisoryData(
+        alias = is_cve(cve)
```
Comment (TG1999) [inline vulnerabilities/importers/mozilla.py:86]:
```suggestion
```

File: vulnerabilities/importers/mozilla.py [hunk 4]
```diff
@@ -146,18 +136,29 @@ def mfsa_id_from_filename(filename):
...
             # pkg is of the form "Firefox ESR 1.21" or "Thunderbird 2.21"
-            name=pkg.rsplit(None, 1)[0],
-            version=pkg.rsplit(None, 1)[1],
-        )
-        for pkg in pkgs
-        if pkg
-    ]
-    return package_urls
+        version = pkg.rsplit(None, 1)[1]
+        name = pkg.rsplit(None, 1)[0]
```
Comment (TG1999) [inline vulnerabilities/importers/mozilla.py:145]:
```suggestion
```

**Review by pombredanne — APPROVED** [review]
LGTM!

**Review by TG1999 — COMMENTED** [review]
some suggestions

**Pull request merged**